tests/Unit/ResearchQueueServiceTest.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Tests\Unit;
+
+use OGame\Models\ResearchQueue;
+use OGame\Services\ObjectService;
+use OGame\Services\ResearchQueueService;
+use Tests\AccountTestCase;
+
+class ResearchQueueServiceTest extends AccountTestCase
```

Comment:
Since this test now extends from AccountTestCase it makes this a `Feature` test as it communicates with the database. 

The goal for the tests in the `tests/Unit` folder is to be able to run without a database dependency and be able to mock all services instead.

So in this case if the new logic requires a database to test properly, could you move the whole file to the `tests/Feature` folder and adjust the namespaces accordingly?

Reply:
I refactored test cases and moved feature tests to proper place. I also added new feature cases to cover discussed use cases. Thanks for pointing out improperly done test cases.